test/e2e/suite-features/external-sd.go

```diff
@@ -238,6 +238,16 @@ spec:
 			// -- validation --
 			upstreamId := PhaseValidateFirstUpstream(s, 1, fqdn, "dns")
 			PhaseValidateRouteAccess(s, upstreamId)
+
+			//Check the status of ApisixUpstream resource
+			upstreamStatus, err := s.GetApisixResourceStatus("httpbin-upstream", "au")
+			assert.Nil(ginkgo.GinkgoT(), err)
+			assert.Equal(ginkgo.GinkgoT(), "Sync Successfully", upstreamStatus.Conditions[0].Message)
```

Comment:
Directly calling these three lines as a function can reduce a lot of repetitive code.

Reply:
Its just three lines and in tests many such lines are repeated a bunch of times. And using function, we will have exactly same assertions for all cases. But here we can add/modify some other status assertions based on a particular test.

Reply:
The asserted message can be used as the parameter of the function, and the advantage of using the same function is that the logic of the assertion can be adjusted in one place in the future. For example, I have an idea to optimize CI, and now there are many sleeps in CI, in fact, the waiting process can be accelerated by polling the CRD state. If we have a unified function, You can easily transform it into this. Don't repeat just because the repeating snippet is small.

Reply:
okay i put it in a function now